sql/core/src/test/scala/org/apache/spark/sql/internal/SQLConfSuite.scala

```diff
@@ -221,6 +221,19 @@ class SQLConfSuite extends QueryTest with SharedSQLContext {
       .sessionState.conf.warehousePath.stripSuffix("/"))
   }
 
+  test("DATA_SOURCE_TABLE_RELATION_CACHE_MAX_SIZE") {
+    assert(spark.conf.get(SQLConf.DATA_SOURCE_TABLE_RELATION_CACHE_MAX_SIZE) === 1000)
+    withSQLConf(SQLConf.DATA_SOURCE_TABLE_RELATION_CACHE_MAX_SIZE.key -> "2000") {
+      assert(spark.conf.get(SQLConf.DATA_SOURCE_TABLE_RELATION_CACHE_MAX_SIZE) === 2000)
+    }
```

Comment:
We can change the value at the runtime, but the changed value does not affect the actual cache size. 

I am thinking if we should move the conf to `StaticSQLConf`, unless we really need to allow users to change it at runtime.